api/src/routes/cart.ts

```diff
@@ -0,0 +1,321 @@
+ *   put:
+ *     summary: Update cart item quantity
+ *     tags: [Cart]
+ *     parameters:
+ *       - in: path
+ *         name: sessionId
+ *         required: true
+ *         schema:
+ *           type: string
+ *         description: Session identifier for the cart
+ *       - in: path
+ *         name: cartItemId
+ *         required: true
+ *         schema:
+ *           type: integer
+ *         description: Cart item ID
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             required:
+ *               - quantity
+ *             properties:
+ *               quantity:
+ *                 type: integer
+ *                 minimum: 1
+ *                 description: New quantity for the item
+ *     responses:
+ *       200:
+ *         description: Item quantity updated successfully
+ *       400:
+ *         description: Invalid quantity
+ *       404:
+ *         description: Cart or cart item not found
+ *   delete:
+ *     summary: Remove item from cart
+ *     tags: [Cart]
+ *     parameters:
+ *       - in: path
+ *         name: sessionId
+ *         required: true
+ *         schema:
+ *           type: string
+ *         description: Session identifier for the cart
+ *       - in: path
+ *         name: cartItemId
+ *         required: true
+ *         schema:
+ *           type: integer
+ *         description: Cart item ID
+ *     responses:
+ *       204:
+ *         description: Item removed from cart successfully
+ *       404:
+ *         description: Cart or cart item not found
+ */
+
+import express from 'express';
+import { getCartRepository } from '../repositories/cartRepo';
+import { NotFoundError, ValidationError } from '../utils/errors';
+
+const router = express.Router();
+
+// Get or create cart with items
+router.get('/:sessionId', async (req, res, next) => {
+  try {
+    const { sessionId } = req.params;
+    const repo = await getCartRepository();
+
+    // Get existing cart or create new one
+    let cart = await repo.getCartBySessionId(sessionId);
+    if (!cart) {
+      cart = await repo.createCart(sessionId);
+    }
+
+    // Get cart items with product details
+    const items = await repo.getCartItems(cart.cartId);
+
+    res.json({
+      cart,
+      items,
+    });
+  } catch (error) {
+    next(error);
+  }
+});
+
+// Add item to cart
+router.post('/:sessionId/items', async (req, res, next) => {
+  try {
+    const { sessionId } = req.params;
+    const { productId, quantity } = req.body;
+
+    // Validate input
+    if (!productId || !quantity) {
+      throw new ValidationError('productId and quantity are required');
+    }
+
+    if (typeof productId !== 'number' || typeof quantity !== 'number') {
+      throw new ValidationError('productId and quantity must be numbers');
+    }
+
+    if (quantity < 1) {
+      throw new ValidationError('quantity must be at least 1');
+    }
```

Comment:
The validation checks `typeof productId !== 'number'` but doesn't validate that the numbers are positive integers. This could allow negative numbers, decimals, Infinity, or NaN to pass validation. While the database CHECK constraint (quantity >= 1) will catch some of these, it's better to validate at the API layer with clearer error messages.

Consider adding integer validation:
```typescript
if (!Number.isInteger(productId) || productId <= 0) {
  throw new ValidationError('productId must be a positive integer');
}
if (!Number.isInteger(quantity) || quantity < 1) {
  throw new ValidationError('quantity must be a positive integer');
}
```
```suggestion
    if (!Number.isInteger(productId) || productId <= 0) {
      throw new ValidationError('productId must be a positive integer');
    }
    if (!Number.isInteger(quantity) || quantity < 1) {
      throw new ValidationError('quantity must be a positive integer');
    }
```